datasets/enriched_web_nlg/enriched_web_nlg.py

```diff
@@ -0,0 +1,189 @@
+        "lid": [ex.get("lid", "") for ex in entry.get("lex", [])],
+        # all of the sequence are within their own 1-element sublist, else the [0]
+        "text": [ex.get("text", "")[0] for ex in entry.get("lex", [])],
+    }
+    res["shape"] = entry.get("shape", "")
+    res["shape_type"] = entry.get("shape_type", "")
+    return res
+
+
+def xml_file_to_examples(filename):
+    tree = ET.parse(filename).getroot()
+    examples = et_to_dict(tree)["benchmark"]["entries"][0]["entry"]
+    return [parse_entry(entry) for entry in examples]
+
+
+class EnrichedWebNlg(datasets.GeneratorBasedBuilder):
+    """The WebNLG corpus"""
+
+    VERSION = datasets.Version("1.5.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="en", description="Enriched English version of the WebNLG data"),
+        datasets.BuilderConfig(name="de", description="Enriched German version of the WebNLG data"),
+    ]
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "category": datasets.Value("string"),
+                "size": datasets.Value("int32"),
+                "eid": datasets.Value("string"),
+                "original_triple_sets": datasets.Sequence(
+                    {"otriple_set": datasets.Sequence(datasets.Value("string"))}
+                ),
+                "modified_triple_sets": datasets.Sequence(
+                    {"mtriple_set": datasets.Sequence(datasets.Value("string"))}
+                ),
+                "shape": datasets.Value("string"),
+                "shape_type": datasets.Value("string"),
+                "lex": datasets.Sequence(
+                    {
+                        "comment": datasets.Value("string"),
+                        "lid": datasets.Value("string"),
+                        "text": datasets.Value("string"),
+                    }
+                ),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            citation=_CITATION,
+            license=_LICENSE,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URL)
+        # Downloading the repo adds the current commit sha to the directory, so we can't specify the directory
+        # name in advance.
+        revision_name = os.listdir(data_dir)[0]
```

Comment:
ideally we should try to avoid doing that

If you force the use of a specific commit it should be fine noe ?